File: 0487-max-consecutive-ones-ii/0487-max-consecutive-ones-ii.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int ans = 0, dp1 = 0, dp0 = 0;
        for(int i = 0; i < nums.size(); ++i) {
            if(nums[i] == 1) {
                ++dp1; ++dp0;
            } else {
                dp1 = dp0+1;
                dp0 = 0;
            }
            ans = max(ans, dp1);
        }
        return ans;
    }
};

File: 0487-max-consecutive-ones-ii/0487-max-consecutive-ones-ii.py
class Solution(object):
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        ans = dp0 = dp1 = 0
        for x in nums:
            if x == 1:
                dp0 += 1
                dp1 += 1
            else:
                dp1 = dp0 + 1
                dp0 = 0
            ans = max(ans, dp1)
        return ans;